### ISSUE TITLE
React Native 0.63.3 File upload gives "Network error" even if the request gives a 200 response.

### ISSUE THREAD

**Issue by dlogvin** [facebook/react-native#30071]
## Description
A normal form submission without file works perfectly fine, but when it comes to file uploads, fetch or axios returns a network error even if the answer from the server is a 200.

## React Native version:
```
System:
    OS: macOS Mojave 10.14.5
    CPU: (4) x64 Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz
    Memory: 87.15 MB / 8.00 GB
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 14.5.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.5 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.7.5 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 13.2, DriverKit 19.0, macOS 10.15, tvOS 13.2, watchOS 6.1
    Android SDK:
      API Levels: 19, 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.1
      System Images: android-29 | Google Play Intel x86 Atom, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom
      Android NDK: 21.3.6528147
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6626763
    Xcode: 11.3.1/11C504 - /usr/bin/xcodebuild
  Languages:
    Java: 12.0.1 - /usr/bin/javac
    Python: 2.7.17 - /Users/daniel/opt/miniconda2/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.3 => 0.63.3
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```
## Steps To Reproduce
1. Make a form with file upload:
```
const data = new FormData()
            data.append('subject_id', this.props.navigation.getParam('id'))
            data.append('name', this.state.title)
            data.append('progress', this.state.progress * 100)
            data.append('description', this.state.description)
            data.append('date', this.state.date)
            if(this.state.image != null){
                data.append('image', {
                    uri: this.state.image,
                    type: 'image/jpeg',
                    name: 'image'
                });
            }

            axios.post('https://site.com/api/auth/createTask', data, {
                headers: {
                    'Authorization': access,
                    "Content-Type": "multipart/form-data"
                },
            }).then(res => {
                this.setState({ secondaryLoader: false });
                this.props.navigation.navigate('ViewHW', { id: res.data.id })
            })
```
2. Use Android Simulator or Physical device, on build or release verison (does not really matter)
3. Submit the form
4. Cry because it doesn't work

## Expected Results
For the form to perform the `.then`

## Snack, code example, screenshot, or link to a repository:
Already given above.

**Comment by dlogvin** [comment]
@safaiyeh

**Comment by maxgr0** [comment]
@dlogvin got the same issue using react-native `0.63.2`. But only on android!

**Comment by dlogvin** [comment]
> @dlogvin got the same issue using react-native `0.63.2`. But only on android!

Yes, this is specifically for android, in iOS it works perfectly fine

**Comment by maxgr0** [comment]
> > @dlogvin got the same issue using react-native `0.63.2`. But only on android!
> 
> Yes, this is specifically for android, in iOS it works perfectly fine

Everything > Android 9 is working here. Seems to be an issue related to older versions of android. Setting a network configuration in `android/xml` and `AndroidManifest` does not work either.

**Comment by dlogvin** [comment]
@maxgr0 has nothing to do with that network config, what I think you tried to do was to allow http, but in this case, it has nothing to do with this. This is about sending POST requests that have a file inside them. This is a huge issue.

**Comment by maxgr0** [comment]
> @maxgr0 has nothing to do with that network config, what I think you tried to do was to allow http, but in this case, it has nothing to do with this. This is about sending POST requests that have a file inside them. This is a huge issue.

As this is https, that would not solve the issue, I know. I thought it was related to certificates which are not right so I allowed untrusted certificates. But that did not work 🤣

**Comment by appidea** [comment]
Isn't it still issue about Flipper?
I had identically the same problem with react-native@0.63.2.

Disabling Flipper helped.

https://github.com/facebook/react-native/issues/28551#issuecomment-610652110

**Comment by ravirajn22** [comment]
Is this issue happening in release and does disabling flipper help? can you check. As @appidea said it could be due to flipper integration. @dlogvin @maxgr0

**Comment by dlogvin** [comment]
@appidea @ravirajn22 Already tried, still having the issue. I was asked in that issue to open a new one.

**Comment by ravirajn22** [comment]
@dlogvin I checked in my app, tried uploading an image file using fetch using react-native v0.63.3, android v7.1, v9. It is working as expected. 

Try,
1. Clear android build directory, clear gradle cache folders
2. Remove node_modules
3. Try uploading the file only via form data, don't add append any other data. Try if it works.
4. Visit your js code again

**Comment by dlogvin** [comment]
> @dlogvin I checked in my app, tried uploading an image file using fetch using react-native v0.63.3, android v7.1, v9. It is working as expected. 
> 
> Try,
> 1. Clear android build directory, clear gradle cache folders
> 2. Remove node_modules
> 3. Try uploading the file only via form data, don't add append any other data. Try if it works.
> 4. Visit your js code again

I have other forms in my app that have file upload, they work too, only one is giving a network error, but it is a plain "network error". I already trield all the above. The form used to work before I upgraded to 0.62

**Comment by dhavaljivani** [comment]
I have same issue in my react native application, I have two forms with the image in my app one is working perfectly but the second form giving a network error.
I have already tried all the solution but not working I have also update react-native to 0.63.3

**Comment by dlogvin** [comment]
> I have same issue in my react native application, I have two forms with the image in my app one is working perfectly but the second form giving a network error.
> I have already tried all the solution but not working I have also update react-native to 0.63.3

Hello! I found the only solution available:

Create a NEW react-native project from scratch, and transfer your files to that one (the components) and app.js, dont forget to install all the libraries you have on your current project in the new one and to apply all the Android configuration you had to the new one as well as on xcode.

**Comment by dhavaljivani** [comment]
I have created a new react-native project but still the same error

TypeError: Network request failed
TypeError: Network request failed
    at http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:26084:20
    at http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:30672:21
    at _callTimer (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:30588:9)
    at Object.callTimers (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:30796:9)
    at MessageQueue.__callFunction (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2765:31)
    at http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2497:17
    at MessageQueue.__guard (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2719:13)
    at MessageQueue.callFunctionReturnFlushedQueue (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2496:14)
    at http://localhost:8081/debugger-ui/debuggerWorker.cff11639.js:4:902

**Comment by dlogvin** [comment]
> @dlogvin
> I have created a new react-native project but still the same error
> 
> TypeError: Network request failed
> TypeError: Network request failed
>     at http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:26084:20
>     at http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:30672:21
>     at _callTimer (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:30588:9)
>     at Object.callTimers (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:30796:9)
>     at MessageQueue.__callFunction (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2765:31)
>     at http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2497:17
>     at MessageQueue.__guard (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2719:13)
>     at MessageQueue.callFunctionReturnFlushedQueue (http://localhost:8081/index.bundle?platform=android&dev=true&minify=false:2496:14)
>     at http://localhost:8081/debugger-ui/debuggerWorker.cff11639.js:4:902

Can I see the whole form and your package.json?

**Comment by cokealmonacid** [comment]
Any updates on this? same issue here

**Comment by navicstein** [comment]
Am having still having issues with this stuff although am using expo

**Comment by joaom182** [comment]
Same issue here. Android and iOS

**Comment by viniengelage** [comment]
Any updates on this? Same issue here

**Comment by nyenye** [comment]
Here with same issue. At first my issue was that when attempting to upload a file using a multipart-form, the request didn't reach the server, or if it did, the file was always undefined. And like @appidea said, disabling Flipper, solved the issue. The file was being uploaded as expected. But what I didn't notice then, was that the request would always throw a Network Error, even though the response code is 201.

React Native version: 0.63.3
Request library: Axios

**Comment by tboerc** [comment]
This still a issue on RN 0.63.3 even with Flipper disabled on Android. The request and API works fine and returns status 200. But the app gives me a blank Network Error. Create a new project and install all dependencies is not a solution to me.

**Comment by cokealmonacid** [comment]
@nyenye  Same situation, making all the suggested changes but nothing seems to work, the request always throw Network Error

**Comment by Siddharth2212** [comment]
This solution works perfectly fine: 
1. Add this at the end of your file android/gradle.properties: FLIPPER_VERSION=0.52.1
2. Make sure you have these two lines in the dependencies section in android/app/build.gradle:  
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }

    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }

**Comment by ahayaSKBINCN** [comment]
Here with same issus. I did all the things with flipper, but nothing change. The Network Error occured after i upload img. And I have two image uploader. one worked fine but another got  this error. I tried to find difference between them. And at last I find 
the error one used image cropper and when I structure the FormData,  I append the cropRect to the FormData. And after I stringified the cropRect, all the things be fine.
